test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/ListPackageTests.cs

```diff
@@ -156,6 +163,123 @@ public void BasicListPackage_OutputFormat_BadInput_Parsing_Fails(string outputFo
+            mockedCredentialService.Verify(x => x.TryGetLastKnownGoodCredentialsFromCache(It.IsAny<Uri>(), It.IsAny<bool>(), out It.Ref<ICredentials>.IsAny), Times.Exactly(3));
+
+            static void SetupCredentialServiceMock(Mock<ICredentialService> mockedCredentialService, NetworkCredential expectedCredentials, Uri packageSourceUri)
+            {
+                NetworkCredential cachedCredentials = default;
+                mockedCredentialService.SetupGet(x => x.HandlesDefaultCredentials).Returns(true);
+                // Setup GetCredentialsAsync mock
+                mockedCredentialService
+                    .Setup(x => x.GetCredentialsAsync(packageSourceUri, It.IsAny<IWebProxy>(), CredentialRequestType.Unauthorized, It.IsAny<string>(), It.IsAny<CancellationToken>()))
+                    .ReturnsAsync(() =>
+                    {
+                        cachedCredentials = expectedCredentials;
+                        return cachedCredentials;
+                    });
+                // Setup TryGetLastKnownGoodCredentialsFromCache mock
+                mockedCredentialService
+                    .Setup(x => x.TryGetLastKnownGoodCredentialsFromCache(packageSourceUri, It.IsAny<bool>(), out It.Ref<ICredentials>.IsAny))
+                    .Returns((Uri sourceUri, bool isProxyRequest, out ICredentials outCredentials) =>
+                    {
+                        outCredentials = cachedCredentials;
+                        return outCredentials != null;
+                    });
+            }
+
+            static async Task RestoreProjectsAsync(SimpleTestPathContext pathContext, SimpleTestProjectContext projectA, SimpleTestProjectContext projectB)
+            {
+                var settings = Settings.LoadDefaultSettings(Path.GetDirectoryName(pathContext.NuGetConfig), Path.GetFileName(pathContext.NuGetConfig), null);
+                var packageSourceProvider = new PackageSourceProvider(settings);
+
+                var sources = packageSourceProvider.LoadPackageSources();
+                var fallbackFolders = (IList<string>)SettingsUtility.GetFallbackPackageFolders(settings);
+                var globalPackagesFolder = SettingsUtility.GetGlobalPackagesFolder(settings);
+
+                await RestoreProjectAsync(settings, pathContext, projectA, sources, fallbackFolders, globalPackagesFolder);
+                await RestoreProjectAsync(settings, pathContext, projectB, sources, fallbackFolders, globalPackagesFolder);
+            }
+
+            static async Task RestoreProjectAsync(ISettings settings,
+                SimpleTestPathContext pathContext,
+                SimpleTestProjectContext project,
+                IEnumerable<PackageSource> packageSources,
+                IList<string> fallbackFolders,
+                string globalpackagesFolder)
+            {
+                project.Sources = packageSources;
```

Comment:
Aren't there some test helpers utils that set these up?
In particular https://github.com/NuGet/NuGet.Client/blob/dev/test/TestUtilities/Test.Utility/Commands/ProjectTestHelpers.cs#L284-L291

Reply:
If I replace all the lines from 271 to 274 with the following 2 lines then the test is passing. Otherwise, test fails with an error message saying `value is null` for Source property.

```cs
project.Sources = packageSources;
var packageSpec = ProjectTestHelpers.WithSettingsBasedRestoreMetadata(project.PackageSpec, settings);
```

Reply:
Sources are not different. It is just that `Project.Sources` is null. Hence, accessing `Project.PackageSpec` throws a null reference exception because existing implementation has `Sources.ToList()` https://github.com/NuGet/NuGet.Client/blob/dev/test/TestUtilities/Test.Utility/SimpleTestSetup/SimpleTestProjectContext.cs#L261. 

I added a null check and changed the test code just to invoke  `ProjectTestHelpers.WithSettingsBasedRestoreMetadata(project.PackageSpec, settings);` and the tests passed locally and pushed the commit.